### ISSUE TITLE
Run the tests suite from within the GUI version of GAMA.

### ISSUE THREAD

**Issue by AlexisDrogoul** [gama-platform/gama#2265]
A more robust test infrastructure has been built into GAMA recently, allowing both GUI and headless runs of `test` experiments, which can be defined in each plugin in a folder named `tests`,  are conveniently grouped into a new directory in the navigator and can be all executed at once using a new headless command (see #2259). 

One of the remaining pieces would be to let the modeler launch this tests suite from the user interface as well and, why not, have it run automatically when an update is performed, in order to verify that there are no regressions, changes or odd behaviours introduced in the update. 

What would be nice is:

- adding a contextual menu item to the 'Test models' directory to run all the tests at once.
- automatically recognising the presence of a 'tests' folder in a GAML project so that modellers can maintain their own crucial tests (depending, for instance, on their dataset in the project, or importing their models) and having executed as well
- proposing a preference for executing these tests (the ones shipped with GAMA + the ones detected in user projects) at every launch of GAMA, or every day, or every week, or at every update.

Please add new ideas if you have some ! 

### System and version 
Current GIT version

**Comment by AlexisDrogoul** [comment]
Hi,
I have implemented a first version of the tests suite execution in the GUI version of GAMA. When the Tests folder is selected, a button appears above:

![relationtests - users drogoul documents git gama msi gama core tests spatial tests models relation experiment 2017-10-10 21-51-22](https://user-images.githubusercontent.com/579256/31393543-273f685c-ae06-11e7-9a0d-b09dc0847f41.png)

It then invokes headless versions of the experiments (but not the headless plugin itself) and runs them sequentially, displaying the results in the same view as the individual tests: 

![relationtests - users drogoul documents git gama msi gama core tests spatial tests models relation experiment 2017-10-10 21-52-18](https://user-images.githubusercontent.com/579256/31393621-5e6f6dd6-ae06-11e7-8920-b974b4b72315.png)

I must say that things are incredibly smooth and quick (main reason being that very few tests have been designed, probably ... :) ). 

I have now to work on the 2 other enhancements: 

- [ ] running tests defined in user projects
- [ ] proposing a preference to run all tests periodically 

and a new 3rd one:

- [ ] displaying a summary in the tests view to avoid scrolling all the way down or proposing to sort the tests by severity (i.e. aborted > failed > warning > passed > not run).

**Comment by AlexisDrogoul** [comment]
I've given up trying to define a robust periodical run of the tests, given the numerous constraints attached to this kind of things (i.e. not to launch while an experiment runs, etc.). Instead, I've installed a new preference for starting tests at each start of the platform (false by default). 

A new preference also allows to include user tests or not in the general tests (false by default). Only the test experiments present in a `tests` folder inside the projects (next to the models ones, for instance) will be considered (so as to not run long or resource-consuming tests at every restart !). They still can be run individually, of course. 

Finally, I've moved the command to run all tests to the validation view. 

![modeling - test1 tests new gaml - gama runtime 2017-10-14 09-29-25](https://user-images.githubusercontent.com/579256/31571733-95885b16-b0c2-11e7-8497-4f90d3b309f7.png)

And last thing : a contextual menu on tests allow them to show their definition in the editor (or copy their summary in the clipboard).

Will all of that, I consider this issue as closed !
